### DISCUSSION TITLE
[Suggestion] Update gtp5g module to v0.8.5+

### DISCUSSION THREAD

**Discussion by howardswe499** [HewlettPackard/PacketRusher#99] [Ideas]
Device info: RaspberryPi 5 running Raspbian Bookworm Kernel 6.1

Hi @linouxis9 , 

The free5gc/gtp5g module that is currently packaged within PackerRusher is v0.8.2, which does not support linux kernels newer than 5.15, and running PR on a system with newer kernel returns the following error upon PDU session establishment:
```
FATA[0002] [GNB][GTP] Unable to create FAR: invalid argument 
```

gtp5g v0.8.5 now supports up to kernel version >=6.2 (https://github.com/free5gc/gtp5g/releases/tag/v0.8.5). I've tried installing the gtp5g module separately (albeit it was version v0.8.6), and it seems to be compatible with PackerRusher. This way people can avoid having to downgrade their kernel.

Thank you for your attention.

Best Regards,
Howard

**Comment by linouxis9** [comment]
Hi @howardswe499,

Thank you for the suggestion! I'll backport the latest GTP5G changes to PacketRusher.

Cheers,
Valentin

**Reply by linouxis9** [reply]
Thanks a lot @howardswe499, PR is here: https://github.com/HewlettPackard/PacketRusher/pull/101

BTW, were you able to use PacketRusher with either the Nokia CMM or CMU product?

Thanks!

Cheers,
Valentin

**Reply by howardswe499** [reply]
Hi @linouxis9 ,

Thanks for your prompt fix👍Unfortunately I've run into some technical difficulties with the CMU I'm testing, so I can't tell you if it works right now. I'll update you if it works.

BR
Howard

**Reply by linouxis9** [reply]
Great, thanks a lot :-)
